sympy/tensor/array/array_comprehension.py

```diff
@@ -0,0 +1,121 @@
+from __future__ import print_function, division
+import functools
+from sympy.core.sympify import sympify
+from sympy.core.expr import Expr
+from sympy.core import Basic
+from sympy.core.compatibility import Iterable
+from sympy.tensor.array import MutableDenseNDimArray, ImmutableDenseNDimArray
+from sympy import  Symbol
+from sympy.core.sympify import sympify
+from sympy.core.compatibility import Iterable
+from sympy.core.numbers import Integer
+
+class ArrayComprehension(Basic):
```

Comment:
This class is missing the `.shape` property, and the `.rank()` function as well. It should be compatible with the arrays.